python/cudf/cudf/pylibcudf_tests/test_source_info.py

```diff
@@ -0,0 +1,69 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+import io
+
+import pytest
+
+import cudf._lib.pylibcudf as plc
+
+
+@pytest.mark.parametrize(
+    "source", ["a.txt", b"hello world", io.BytesIO(b"hello world")]
+)
+def test_source_info_ctor(source, tmp_path):
+    if isinstance(source, str):
+        file = tmp_path / source
+        file.write_bytes("hello world".encode("utf-8"))
+        source = str(file)
+
+    plc.io.SourceInfo([source])
+
+    # TODO: test contents of source_info buffer is correct
+    # once buffers are exposed on python side
```

Comment:
Accessing the buffers of the SourceInfo is a bit annoying right now, due to the host_buffer deprecation, and because the host_span bindings are incomplete.

This is probably better done after the host_buffer -> host_span deprecation adjustment.